server/build_event_protocol/build_status_reporter/build_status_reporter.go

```diff
@@ -45,6 +45,7 @@ func NewBuildStatusReporter(env environment.Env, invocationID string) *BuildStat
 }
 
 func (r *BuildStatusReporter) ReportStatusForEvent(ctx context.Context, event *build_event_stream.BuildEvent) {
+	shouldReportStatusPerTest := r.shouldReportStatusPerTest()
```

Comment:
What do you think about throwing a bool called "reportStatusPerTest" in the BuildStatusReporter struct and initializing it at BuildStatusReporter construction time? That way we're not having to re-reference the config file on every event.